packages/oidc-client/src/lib/oidc.api.ts

```diff
@@ -20,6 +22,159 @@ export const oidcApi = createApi({
   reducerPath: 'oidc',
   baseQuery: fetchBaseQuery(),
   endpoints: (builder) => ({
+    authorizeFetch: builder.mutation<AuthorizeSuccessResponse, { url: string }>({
+      queryFn: async ({ url }, api, _, baseQuery) => {
+        const { requestMiddleware, logger } = api.extra as Extras;
+
+        const request: FetchArgs = {
+          url,
+          method: 'POST',
+          credentials: 'include',
+          headers: {
+            Accept: 'application/json',
+          },
+        };
+
+        logger.debug('OIDC authorize API request', request);
+
+        const response = await initQuery(request, 'authorize')
+          .applyMiddleware(requestMiddleware)
+          .applyQuery(async (req: FetchArgs) => await baseQuery(req));
+
+        if (response.error) {
+          const responseError = response.error;
+
+          // If the details field is present in data, use it to create a more specific error response
+          if (
+            responseError.data &&
+            typeof responseError.data === 'object' &&
+            'details' in responseError.data &&
+            Array.isArray(responseError.data.details)
+          ) {
+            logger.debug('Error in authorize response', responseError);
+
+            const details = responseError.data.details[0] as {
+              code: string;
+              message: string;
+            };
+
+            response.error = {
+              status: responseError.status,
+              statusText: 'AUTHORIZE_ERROR',
+              data: {
+                error: details.code,
+                error_description: details.message,
+                type: 'auth_error',
+              },
+            } as FetchBaseQueryError;
```

Comment:
This technically isn't a FetchBaseQueryError because `statusText` is a an added property. For it to be a FetchBaseQueryError it would have to be something like:
```
response.error = {
              status: responseError.status,
              data: {
                error: details.code,
                error_description: details.message,
                type: 'auth_error',
              },
              error: details.message,
            };
```

Reply:
You are correct. I'm intentionally doing this for reasons that are not easily articulated in a comment. Let's discuss this when we discuss errors on our upcoming call.